Source/GUI/VCL/GUI_Preferences.cpp

```diff
@@ -370,12 +389,16 @@ void __fastcall TPreferencesF::CB_InscrireShellClick(TObject *Sender)
 //---------------------------------------------------------------------------
 void __fastcall TPreferencesF::CB_InscrireShell_FolderClick(TObject *Sender)
 {
+    // Cannot be changed on Windows 11

```

Comment:
Isn't there a way in the DLL to see if it is a file or a folder and show the MediaInfo option depending on the setting?

Reply:
No idea. You can try implement it later but we cannot perform operations that take too long there or else Explorer menu may hang. Also what if user highlights combination of files and folders?

Reply:
> You can try implement it later but we cannot perform operations that take too long there or else Explorer menu may hang.

I don't get there how the selection is done depending on the file extension. If we check the file extension we could check if it is a folder.
But low priority, I can accept such limitation for a first shot.

> Also what if user highlights combination of files and folders?

Same as if files are with a supported extension or not, IMO. Checking current Win10 behavior it is hidden if one of the selected items is not supported.

Reply:
> At the moment it is shown for all file types.

I am reluctant to that. People will complain.
Is it difficult to add a list?
Based on [this list](https://github.com/MediaArea/MediaInfo/blob/master/Source/Install/MediaInfo_Extensions.nsh).

Reply:
Not sure if things like `audio` and `video` are allowed.

Reply:
> Not sure if things like audio and video are allowed.

It would be even better, with the difficulty that some extensions are not recognized by Windows as audio or video.

Reply:
https://learn.microsoft.com/en-us/uwp/schemas/appxpackage/uapmanifestschema/element-desktop5-itemtype

Here are the docs. Looks like no `audio` and `video` allowed. So need to make sure every audio and video extension is manually specified.

Reply:
OK, so nee to fill all supported extensions.
And I see the special name `Directory` ;-).

Reply:
Done with this. Generated with AI since it will be tedious to write that much. Not yet tested. Hope AI did not make any mistakes and that it works.